kafka-client/src/main/java/dev/responsive/db/CassandraFactSchema.java

```diff
@@ -70,17 +75,31 @@ public CassandraFactSchema(final CassandraClient client) {
   }
 
   @Override
-  public void create(final String tableName) {
-    LOG.info("Creating data table {} in remote store.", tableName);
-    client.execute(SchemaBuilder
+  @CheckReturnValue
+  public SimpleStatement create(final String tableName, Optional<Duration> ttl) {
+    LOG.info("Creating fact data table {} in remote store.", tableName);
+    final CreateTable createTable = SchemaBuilder
         .createTable(tableName)
         .ifNotExists()
         .withPartitionKey(ROW_TYPE.column(), DataTypes.TINYINT)
         .withPartitionKey(DATA_KEY.column(), DataTypes.BLOB)
         .withColumn(DATA_VALUE.column(), DataTypes.BLOB)
-        .withColumn(OFFSET.column(), DataTypes.BIGINT)
-        .build()
-    );
+        .withColumn(OFFSET.column(), DataTypes.BIGINT);
+
+    if (ttl.isPresent()) {
+      final int ttlSeconds = Math.toIntExact(ttl.get().getSeconds());
+      // 20 is a magic number recommended by scylla for the number of buckets
+      final Duration compactionWindow = Duration.ofSeconds(ttlSeconds / 20);
+      return createTable.withDefaultTimeToLiveSeconds(ttlSeconds)
+          .withCompaction(
+              SchemaBuilder.timeWindowCompactionStrategy()
+                  .withCompactionWindow(
+                      compactionWindow.toMinutes(),
```

Comment:
uh...are we converting from Duration to `int (seconds)` back to Duration and then finally to `int (minutes)`??

Reply:
the TTL setting needs to be in settings (see `withDefaultTimeToLiveSeconds`) while the TWCS window needs to be in either MINUTES, HOURS or DAYS.